runtime/j9vm/jvm.c

```diff
@@ -553,6 +558,70 @@ captureCommandLine(void)
 		}
 		close(fd);
 	}
+#elif defined(OSX) /* defined(AIXPPC) */
+	int argmax = 0;
+	size_t length = 0;
+	int mib[3];
+
+	/* query the argument space limit */
+	mib[0] = CTL_KERN;
+	mib[1] = KERN_ARGMAX;
+	length = sizeof(argmax);
+	if (0 == sysctl(mib, 2, &argmax, &length, NULL, 0)) {
+		char *buffer = malloc(argmax);
+
+		if (NULL != buffer) {
+			int argc = 0;
+			int pid = getpid();
+
+			/* query the argument count */
+			mib[0] = CTL_KERN;
+			mib[1] = KERN_PROCARGS2;
+			mib[2] = pid;
+			length = argmax;
+			if (0 == sysctl(mib, 3, buffer, &length, NULL, 0)) {
+				memcpy(&argc, buffer, sizeof(argc));
+
+				/* retrieve the arguments */
+				mib[0] = CTL_KERN;
+				mib[1] = KERN_PROCARGS;
+				mib[2] = pid;
+				length = argmax;
+				if (0 == sysctl(mib, 3, buffer, &length, NULL, 0)) {
+					char *cursor = buffer;
+					char *start = NULL;
+					char *limit = buffer + length;
+
+					for (; (cursor < limit) && ('\0' != *cursor); ++cursor) {
+						/* skip past the program path */
+					}
+
+					for (; (cursor < limit) && ('\0' == *cursor); ++cursor) {
+						/* skip past the padding before the first argument */
+					}
+
+					/* remember the start of the first argument */
+					start = cursor;
```

Comment:
Other platforms include the program path in the environment variable.

Reply:
This is consistent with other platforms. Here 'first argument' is `argv[0]` which is normally the program name/path.

Reply:
Fixed in e5003acf8fa36b8c635f0969521c0d658b9bad25.